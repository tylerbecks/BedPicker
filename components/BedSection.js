import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const BedSection = ({ type, name, guests }) => (
  <View>
    <Text style={styles.title}>{type}</Text>
    <Text>{name}</Text>
    {guests.map(guest => (
      <Text>{guest}</Text>
    ))}
  </View>
);

export default BedSection;

const styles = StyleSheet.create({
  title: {
    fontWeight: 'bold',
    fontSize: 24
  }
});
