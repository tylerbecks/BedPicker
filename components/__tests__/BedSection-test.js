import 'react-native';
import React from 'react';
import BedSection from '../BedSection';
import renderer from 'react-test-renderer';

it('renders when guests have photos', () => {
  const guests = [
    {
      name: 'foo',
      photo: {}
    },
    {
      name: 'bar',
      photo: {}
    }
  ];

  expect(
    renderer.create(<BedSection name="baz" guests={guests} />).toJSON()
  ).toMatchSnapshot();
});

it('renders when guests have no photo', () => {
  const guests = [
    {
      name: 'foo',
      photo: null
    }
  ];

  expect(
    renderer.create(<BedSection name="baz" guests={guests} />).toJSON()
  ).toMatchSnapshot();
});
