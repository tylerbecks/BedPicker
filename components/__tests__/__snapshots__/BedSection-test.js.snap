// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when guests have no photo 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 12,
      }
    }
  >
    baz
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 20,
        "paddingVertical": 8,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 15,
          "height": 30,
          "marginRight": 10,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      foo
    </Text>
  </View>
</View>
`;

exports[`renders when guests have photos 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 12,
      }
    }
  >
    baz
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 20,
        "paddingVertical": 8,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "borderRadius": 15,
          "height": 30,
          "marginRight": 10,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      foo
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 20,
        "paddingVertical": 8,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "borderRadius": 15,
          "height": 30,
          "marginRight": 10,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      bar
    </Text>
  </View>
</View>
`;
